### DISCUSSION TITLE
Bedrock sdk missing api support for outputdataconfig on InvokeModel?

### DISCUSSION THREAD

**Discussion by EZWrighter** [aws/aws-sdk-java#3166] [Q&A]
Trying to use SDK to call bedrock model TEXT_VIDEO with amazon.nova-reel-v1:0 and cannot find the API to set the OutputDataConfig on the request or during the call to InvokeModel() on the AmazonBedrockRuntime.  Some way to bypass this issue with InvokeModelRequest setup?

**Comment by debora-ito** [comment]
There isn't a `OutputDataConfig` data type in the BedrockRuntime client, so I don't think it's possible to provide a custom output config at invocation time. 

`OutputDataConfig` is used in the [CreateModelCustomizationJobRequest](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/bedrock/model/CreateModelCustomizationJobRequest.html) in the Bedrock client.

**Reply by EZWrighter** [reply]
Interesting...thanks, I will check that out.  All of the other APIs, even sdk v2 uses the outputDataConfig at invocation.

**Reply by debora-ito** [reply]
> All of the other APIs, even sdk v2 uses the outputDataConfig at invocation.

Which APIs would those be, can you point to a documentation link?

**Comment by EZWrighter** [comment]
The only real Bedrock Nova Reel example I found:
https://docs.aws.amazon.com/nova/latest/userguide/video-gen-code-examples.html
Example of using other Bedrock models in java with outputDataConfig in
invocation:
https://docs.aws.amazon.com/bedrock/latest/userguide/batch-inference-example.html

Maybe you have a better example?  For me the AWS documentation across the
board is such a mess it's hard to find anything.  But I also try not to
spend my days wading through docs on AWS :-P

**Reply by debora-ito** [reply]
Got it, thank you for the links. 

Sorry, in my previous comment I searched for the `OutputDataConfig` class, but the data type for the BedrockRuntime StartAsyncInvoke API is called `AsyncInvokeOutputDataConfig`. Here's the Javadoc for the StartAsyncInvoke request builder: https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/bedrockruntime/model/StartAsyncInvokeRequest.Builder.html

```Java
    StartAsyncInvokeRequest request = StartAsyncInvokeRequest.builder()
        .modelId("amazon.nova-reel-v1:0")
        .modelInput(MODELINPUT)
        .outputDataConfig(AsyncInvokeOutputDataConfig.builder()
            .s3OutputDataConfig(AsyncInvokeS3OutputDataConfig.builder().s3Uri("s3://my-bucket").build())
            .build())
        .build();

    StartAsyncInvokeResponse response = client.startAsyncInvoke(request);
```

**Comment by EZWrighter** [comment]
Ahh, that is awesome...thanks, didn't expect that Class name :-D  I will
give it a try.

**Comment by EZWrighter** [comment]
Also strange that only the async request has this configuration during the
invoke.  I would expect the sync api to match that functionality.

Eric
